django/db/models/expressions.py

```diff
@@ -866,6 +895,112 @@ def as_sql(self, compiler, connection, template=None, case_joiner=None, **extra_
         return sql, sql_params
 
 
+class SubQuery(Expression):
+    """
+    An explicit subquery. It may contain OuterRef() references to the outer
+    query, which will be resolved when it is applied to that query.
+    """
+    template = '(%(subquery)s)'
+
+    def __init__(self, subquery, output_field=None, **extra):
+        self.subquery = subquery.all()
+        self.extra = extra
+        if output_field is None and len(self.subquery.query.select) == 1:
```

Comment:
Interesting that this doesn't catch the `SubQuery(aggregation_qs)` case where I had to define the output_field. I guess it's because the select list is longer (aggregation + grouping column), but another property keeps track of what we're actually returning? It'd be really useful to correctly get the output_field for aggregated subqueries too.